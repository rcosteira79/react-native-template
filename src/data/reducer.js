
import { combineReducers } from 'redux';
import { reducer as usersReducer } from './users/reducer';

export const dataReducer = combineReducers({
    users: usersReducer,
});
