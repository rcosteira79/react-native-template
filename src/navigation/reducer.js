import { NavigationActions } from 'react-navigation';
import { AppNavigator } from '../navigation';

export const initialState = AppNavigator.router.getStateForAction(
	AppNavigator.router.getActionForPathAndParams('Splash')
);

export const navigationReducer = (state = initialState, action) => {
	let newState;
	switch (action.type) {
		case 'Login':
			newState = AppNavigator.router.getStateForAction(
				NavigationActions.navigate({ routeName: 'Login' }),
				state
			);
			break;
		case 'Main':
			newState = AppNavigator.router.getStateForAction(
				NavigationActions.navigate({ routeName: 'Main' }),
				state
			);
			break;
		default:
			newState = AppNavigator.router.getStateForAction(action, state);
			break;
	}

	return newState || state;
};
