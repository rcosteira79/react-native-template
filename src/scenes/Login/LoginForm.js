import dismissKeyboard from 'react-native/Libraries/Utilities/dismissKeyboard';
import React, { Component } from 'react';

import {
	View,
	TouchableOpacity,
	TouchableWithoutFeedback,
	Text,
	StatusBar,
	ActivityIndicator,
} from 'react-native';

import EStyleSheet from 'react-native-extended-stylesheet';

import * as session from '../../services/session';
import * as api from '../../services/api';

import { FloatingLabelInput } from '../../components';

class LoginForm extends Component {
	constructor(props) {
		super(props);

		this.initialState = {
			isLoading: false,
			error: null,
			email: '',
			password: '',
		};

		this.state = this.initialState;
	}

	onPressLogin() {
		this.setState({
			isLoading: true,
			error: '',
		});

		dismissKeyboard();
		session.login(this.state.email, this.state.password)
			.then(() => {
				console.log('authenticated');
			})
			.catch((exception) => {
				// Displays only the first error message
				const error = api.exceptionExtractError(exception);
				this.setState({
					isLoading: false,
					...(error ? { error } : {}),
				});

				if (!error) {
					throw exception;
				}
			});
	}

	setInputReference = (input) => { this.passwordInput = input; };
	handleEmailChange = (email) => this.setState({ email });
	handlePasswordChange = (password) => this.setState({ password });
	focusPassword = () => this.passwordInput.focus();

	render() {
		return (
			<TouchableWithoutFeedback
				onPress={dismissKeyboard}
			>
				<View style={styles.container}>
					<StatusBar hidden />

					<View style={styles.inputContainer}>
						<FloatingLabelInput
							autoCapitalize="none"
							autoCorrect={false}
							keyboardType="email-address"
							returnKeyType="next"
							label="Email"
							value={this.state.email}
							onChangeText={this.handleEmailChange}
							onSubmitEditing={this.focusPassword}
						/>
					</View>

					<View style={styles.inputContainer}>
						<FloatingLabelInput
							label="Password"
							secureTextEntry
							returnKeyType="go"
							value={this.state.password}
							ref={this.setInputReference}
							onChangeText={this.handlePasswordChange}
							onSubmitEditing={this.focusPassword}
						/>
					</View>

					{this.state.isLoading ? (
						<View style={styles.spinnerStyle}>
							<ActivityIndicator
								size={'large'}
								color={EStyleSheet.value('$colorPrimary')}
							/>
						</View>
					) : (
							<TouchableOpacity
								onPress={() => this.onPressLogin()}
								style={styles.loginButtonContainer}
							>
								<Text style={styles.loginButtonText}>
									LOGIN
								</Text>
							</TouchableOpacity>
						)}

					<TouchableOpacity style={styles.registerButtonContainer}>
						<Text style={styles.registerButtonText}>
							Register
						</Text>
					</TouchableOpacity>
				</View>
			</TouchableWithoutFeedback>
		);
	}
}

const styles = EStyleSheet.create({
	container: {
		paddingVertical: 70,
		paddingHorizontal: 70,
	},

	inputContainer: {
		flex: 1,
		marginBottom: 60,
		flexDirection: 'row',
		alignItems: 'center',
		justifyContent: 'center',
	},
	iconContainer: {
		flex: 1,
		marginRight: 20,
	},
	icon: {
		height: 40,
		width: 40,
		resizeMode: 'contain',
	},
	input: {
		flex: 4,
		height: 40,
		color: '$colorPrimary',
	},
	spinnerStyle: {
		marginTop: 40,
		justifyContent: 'center',
		alignItems: 'center',
		paddingVertical: 15,
	},
	loginButtonContainer: {
		marginTop: 40,
		backgroundColor: '$colorAccent',
		paddingVertical: 15,
	},
	loginButtonText: {
		fontSize: 18,
		textAlign: 'center',
		color: 'white',
		fontWeight: 'bold',
	},
	registerButtonContainer: {
		paddingVertical: 15,
	},
	registerButtonText: {
		fontSize: 18,
		textAlign: 'center',
		color: '$colorText',
		fontWeight: 'bold',
	},
});

export default LoginForm;
