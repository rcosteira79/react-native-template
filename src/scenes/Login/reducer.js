import {
  EMAIL_CHANGED,
  PASSWORD_CHANGED,
  LOGIN,
  LOGIN_SUCCESS,
  LOGIN_FAIL
} from './actionTypes';

export const initialState = {
  isLoading: false,
  email: '',
  password: '',
  validEmail: false,
  validPassword: false,
  error: ''
};

export const loginReducer = (state = initialState, action) => {
  switch (action.type) {
    case EMAIL_CHANGED:
    case PASSWORD_CHANGED:
      return { ...state, ...action.payload };
    case LOGIN:
      return { ...state, isLoading: true };
    case LOGIN_SUCCESS:
      return { ...state, ...initialState };
    case LOGIN_FAIL:
      return { ...state, error: action.error, isLoading: false };
    default:
      return state;
  }
};
