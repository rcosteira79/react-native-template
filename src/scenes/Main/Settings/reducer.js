export const initialState = {};

export const settingsReducer = (state = initialState, action) => {
  switch (action.type) {
    default:
      return state;
  }
};
