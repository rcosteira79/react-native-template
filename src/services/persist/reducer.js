import * as actionTypes from './actionTypes';

export const initialState = {
    isHydrated: false,
};

export function persistReducer(state = initialState, action) {
    switch (action.type) {
    case actionTypes.UPDATE:
        return action.payload;
    default:
        return state;
    }
}
