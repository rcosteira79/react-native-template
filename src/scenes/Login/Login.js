import React from 'react';
import { View, Text, Image } from 'react-native';
import EStyleSheet from 'react-native-extended-stylesheet';
import LoginForm from './LoginForm';

const logo = require('../../resources/images/react-native-symbol.png');

const login = () => {
	return (
		<View style={styles.layout}>
			<View style={styles.header}>
				<Text style={styles.title}>
					Hello
				</Text>
				<View>
					<Image
						source={logo}
						style={styles.picture}
					/>
				</View>
			</View>
			<View style={styles.loginForm}>
				<LoginForm />
			</View>
		</View>
	);
};

const styles = EStyleSheet.create({
	layout: {
		backgroundColor: 'white',
		flex: 1,
	},
	header: {
		backgroundColor: '$colorAccent',
		flex: 2,
		justifyContent: 'center',
		alignItems: 'center',
	},
	title: {
		color: 'white',
		fontSize: 32,
		fontWeight: 'bold',
	},
	picture: {
		marginTop: 10,
		width: 150,
		height: 150,
	},
	loginForm: {
		flex: 3,
	},
});

export default login;
