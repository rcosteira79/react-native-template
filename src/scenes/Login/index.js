import React from 'react';
import { View } from 'react-native';
import EStyleSheet from 'react-native-extended-stylesheet';
import LoginForm from './LoginForm';
import { Card } from '../../components';

const Login = () => {
  return (
    <View style={styles.backgroundStyle}>
      <Card style={styles.cardStyle}>
        <LoginForm />
      </Card>
    </View>
  );
};

const styles = EStyleSheet.create({
  backgroundStyle: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: '$colorPrimary'
  },
  cardStyle: {
    marginTop: 88,
    marginBottom: 87,
    marginLeft: 10,
    marginRight: 10
  }
});

export default Login;
