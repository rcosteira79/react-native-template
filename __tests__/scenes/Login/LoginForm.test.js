import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import * as LoginReducer from '../../../src/scenes/Login/reducer';
import LoginForm from '../../../src/scenes/Login/LoginForm';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

const initialState = {
  login: {
    ...LoginReducer.initialState
  },
  locale: {
    languages: [
      { code: 'pt', active: true },
      { code: 'en', active: false },
      { code: 'fr', active: false }
    ],
    translations: {}
  }
};

let state;

Enzyme.configure({ adapter: new Adapter() });

jest.unmock();

describe('Login Form tests', () => {
  let wrapper;

  beforeEach(() => {
    state = initialState;
  });

  it('renders the Login form', () => {
    wrapper = shallow(<LoginForm />, { context: { store: mockStore(state) } });
    expect(wrapper.dive()).toMatchSnapshot();
  });

  it('Displays the email address', () => {
    state.login.email = 'address@email.com';
    wrapper = shallow(<LoginForm />, { context: { store: mockStore(state) } });
    expect(wrapper.dive()).toMatchSnapshot();
  });

  it('Displays invalid email address', () => {
    state.login.email = 'address@';
    state.login.validEmail = false;
    wrapper = shallow(<LoginForm />, { context: { store: mockStore(state) } });
    expect(wrapper.dive()).toMatchSnapshot();
  });

  it('Displays the password', () => {
    state.login.password = 'password';
    wrapper = shallow(<LoginForm />, { context: { store: mockStore(state) } });
    expect(wrapper.dive()).toMatchSnapshot();
  });

  it('Displays invalid password', () => {
    state.login.password = '';
    state.login.validPassword = false;
    wrapper = shallow(<LoginForm />, { context: { store: mockStore(state) } });
    expect(wrapper.dive()).toMatchSnapshot();
  });

  it('Displays error on invalid login credentials', () => {
    state.login.error = 'Invalid login';
    wrapper = shallow(<LoginForm />, { context: { store: mockStore(state) } });
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
