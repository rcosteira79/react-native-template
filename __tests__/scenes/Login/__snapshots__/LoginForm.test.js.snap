// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Form tests Displays error on invalid login credentials 1`] = `
<TouchableWithoutFeedback
  onPress={[Function]}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "paddingHorizontal": 20,
      }
    }
  >
    <View
      style={
        Object {
          "marginTop": 40,
          "maxHeight": 500,
          "minHeight": 100,
          "width": 375,
        }
      }
    >
      <Image
        resizeMode="contain"
        source={1}
        style={
          Object {
            "flex": 1,
            "height": undefined,
            "width": undefined,
          }
        }
      />
    </View>
    <FloatingLabelInput
      autoCapitalize="none"
      autoCorrect={false}
      blurOnSubmit={false}
      focusedColor="#55C5CE"
      keyboardType="email-address"
      label=""
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      returnKeyType="next"
      style={
        Object {
          "marginBottom": 15,
        }
      }
      unfocusedColor="#444444"
      value="address@"
    />
    <View
      style={
        Object {
          "alignSelf": "stretch",
          "marginBottom": 40,
        }
      }
    >
      <FloatingLabelInput
        blurOnSubmit={true}
        focusedColor="#55C5CE"
        label=""
        onChangeText={[Function]}
        onSubmitEditing={[Function]}
        refField={[Function]}
        returnKeyType="done"
        secureTextEntry={true}
        unfocusedColor="#444444"
        value=""
        withRef={true}
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "flex-start",
            "color": "#f54B5E",
            "fontFamily": "Montserrat-Regular",
            "fontSize": 11,
            "marginTop": 2,
          }
        }
      />
    </View>
    <Button
      enabled={false}
      onPress={[Function]}
      style={
        Object {
          "marginBottom": 30,
        }
      }
    />
    <TouchableOpacity
      activeOpacity={0.2}
      style={
        Object {
          "marginBottom": 50,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#16ADC1",
            "fontFamily": "Montserrat-Regular",
            "fontSize": 11,
            "textAlign": "center",
            "textDecorationLine": "underline",
          }
        }
      />
    </TouchableOpacity>
  </View>
</TouchableWithoutFeedback>
`;

exports[`Login Form tests Displays invalid email address 1`] = `
<TouchableWithoutFeedback
  onPress={[Function]}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "paddingHorizontal": 20,
      }
    }
  >
    <View
      style={
        Object {
          "marginTop": 40,
          "maxHeight": 500,
          "minHeight": 100,
          "width": 375,
        }
      }
    >
      <Image
        resizeMode="contain"
        source={1}
        style={
          Object {
            "flex": 1,
            "height": undefined,
            "width": undefined,
          }
        }
      />
    </View>
    <FloatingLabelInput
      autoCapitalize="none"
      autoCorrect={false}
      blurOnSubmit={false}
      focusedColor="#55C5CE"
      keyboardType="email-address"
      label=""
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      returnKeyType="next"
      style={
        Object {
          "marginBottom": 15,
        }
      }
      unfocusedColor="#444444"
      value="address@"
    />
    <View
      style={
        Object {
          "alignSelf": "stretch",
          "marginBottom": 40,
        }
      }
    >
      <FloatingLabelInput
        blurOnSubmit={true}
        focusedColor="#55C5CE"
        label=""
        onChangeText={[Function]}
        onSubmitEditing={[Function]}
        refField={[Function]}
        returnKeyType="done"
        secureTextEntry={true}
        unfocusedColor="#444444"
        value=""
        withRef={true}
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "flex-start",
            "color": "#f54B5E",
            "fontFamily": "Montserrat-Regular",
            "fontSize": 11,
            "marginTop": 2,
          }
        }
      >
         
      </Text>
    </View>
    <Button
      enabled={false}
      onPress={[Function]}
      style={
        Object {
          "marginBottom": 30,
        }
      }
    />
    <TouchableOpacity
      activeOpacity={0.2}
      style={
        Object {
          "marginBottom": 50,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#16ADC1",
            "fontFamily": "Montserrat-Regular",
            "fontSize": 11,
            "textAlign": "center",
            "textDecorationLine": "underline",
          }
        }
      />
    </TouchableOpacity>
  </View>
</TouchableWithoutFeedback>
`;

exports[`Login Form tests Displays invalid password 1`] = `
<TouchableWithoutFeedback
  onPress={[Function]}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "paddingHorizontal": 20,
      }
    }
  >
    <View
      style={
        Object {
          "marginTop": 40,
          "maxHeight": 500,
          "minHeight": 100,
          "width": 375,
        }
      }
    >
      <Image
        resizeMode="contain"
        source={1}
        style={
          Object {
            "flex": 1,
            "height": undefined,
            "width": undefined,
          }
        }
      />
    </View>
    <FloatingLabelInput
      autoCapitalize="none"
      autoCorrect={false}
      blurOnSubmit={false}
      focusedColor="#55C5CE"
      keyboardType="email-address"
      label=""
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      returnKeyType="next"
      style={
        Object {
          "marginBottom": 15,
        }
      }
      unfocusedColor="#444444"
      value="address@"
    />
    <View
      style={
        Object {
          "alignSelf": "stretch",
          "marginBottom": 40,
        }
      }
    >
      <FloatingLabelInput
        blurOnSubmit={true}
        focusedColor="#55C5CE"
        label=""
        onChangeText={[Function]}
        onSubmitEditing={[Function]}
        refField={[Function]}
        returnKeyType="done"
        secureTextEntry={true}
        unfocusedColor="#444444"
        value=""
        withRef={true}
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "flex-start",
            "color": "#f54B5E",
            "fontFamily": "Montserrat-Regular",
            "fontSize": 11,
            "marginTop": 2,
          }
        }
      >
         
      </Text>
    </View>
    <Button
      enabled={false}
      onPress={[Function]}
      style={
        Object {
          "marginBottom": 30,
        }
      }
    />
    <TouchableOpacity
      activeOpacity={0.2}
      style={
        Object {
          "marginBottom": 50,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#16ADC1",
            "fontFamily": "Montserrat-Regular",
            "fontSize": 11,
            "textAlign": "center",
            "textDecorationLine": "underline",
          }
        }
      />
    </TouchableOpacity>
  </View>
</TouchableWithoutFeedback>
`;

exports[`Login Form tests Displays the email address 1`] = `
<TouchableWithoutFeedback
  onPress={[Function]}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "paddingHorizontal": 20,
      }
    }
  >
    <View
      style={
        Object {
          "marginTop": 40,
          "maxHeight": 500,
          "minHeight": 100,
          "width": 375,
        }
      }
    >
      <Image
        resizeMode="contain"
        source={1}
        style={
          Object {
            "flex": 1,
            "height": undefined,
            "width": undefined,
          }
        }
      />
    </View>
    <FloatingLabelInput
      autoCapitalize="none"
      autoCorrect={false}
      blurOnSubmit={false}
      focusedColor="#55C5CE"
      keyboardType="email-address"
      label=""
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      returnKeyType="next"
      style={
        Object {
          "marginBottom": 15,
        }
      }
      unfocusedColor="#444444"
      value="address@email.com"
    />
    <View
      style={
        Object {
          "alignSelf": "stretch",
          "marginBottom": 40,
        }
      }
    >
      <FloatingLabelInput
        blurOnSubmit={true}
        focusedColor="#55C5CE"
        label=""
        onChangeText={[Function]}
        onSubmitEditing={[Function]}
        refField={[Function]}
        returnKeyType="done"
        secureTextEntry={true}
        unfocusedColor="#444444"
        value=""
        withRef={true}
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "flex-start",
            "color": "#f54B5E",
            "fontFamily": "Montserrat-Regular",
            "fontSize": 11,
            "marginTop": 2,
          }
        }
      >
         
      </Text>
    </View>
    <Button
      enabled={false}
      onPress={[Function]}
      style={
        Object {
          "marginBottom": 30,
        }
      }
    />
    <TouchableOpacity
      activeOpacity={0.2}
      style={
        Object {
          "marginBottom": 50,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#16ADC1",
            "fontFamily": "Montserrat-Regular",
            "fontSize": 11,
            "textAlign": "center",
            "textDecorationLine": "underline",
          }
        }
      />
    </TouchableOpacity>
  </View>
</TouchableWithoutFeedback>
`;

exports[`Login Form tests Displays the password 1`] = `
<TouchableWithoutFeedback
  onPress={[Function]}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "paddingHorizontal": 20,
      }
    }
  >
    <View
      style={
        Object {
          "marginTop": 40,
          "maxHeight": 500,
          "minHeight": 100,
          "width": 375,
        }
      }
    >
      <Image
        resizeMode="contain"
        source={1}
        style={
          Object {
            "flex": 1,
            "height": undefined,
            "width": undefined,
          }
        }
      />
    </View>
    <FloatingLabelInput
      autoCapitalize="none"
      autoCorrect={false}
      blurOnSubmit={false}
      focusedColor="#55C5CE"
      keyboardType="email-address"
      label=""
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      returnKeyType="next"
      style={
        Object {
          "marginBottom": 15,
        }
      }
      unfocusedColor="#444444"
      value="address@"
    />
    <View
      style={
        Object {
          "alignSelf": "stretch",
          "marginBottom": 40,
        }
      }
    >
      <FloatingLabelInput
        blurOnSubmit={true}
        focusedColor="#55C5CE"
        label=""
        onChangeText={[Function]}
        onSubmitEditing={[Function]}
        refField={[Function]}
        returnKeyType="done"
        secureTextEntry={true}
        unfocusedColor="#444444"
        value="password"
        withRef={true}
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "flex-start",
            "color": "#f54B5E",
            "fontFamily": "Montserrat-Regular",
            "fontSize": 11,
            "marginTop": 2,
          }
        }
      >
         
      </Text>
    </View>
    <Button
      enabled={false}
      onPress={[Function]}
      style={
        Object {
          "marginBottom": 30,
        }
      }
    />
    <TouchableOpacity
      activeOpacity={0.2}
      style={
        Object {
          "marginBottom": 50,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#16ADC1",
            "fontFamily": "Montserrat-Regular",
            "fontSize": 11,
            "textAlign": "center",
            "textDecorationLine": "underline",
          }
        }
      />
    </TouchableOpacity>
  </View>
</TouchableWithoutFeedback>
`;

exports[`Login Form tests renders the Login form 1`] = `
<TouchableWithoutFeedback
  onPress={[Function]}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "paddingHorizontal": 20,
      }
    }
  >
    <View
      style={
        Object {
          "marginTop": 40,
          "maxHeight": 500,
          "minHeight": 100,
          "width": 375,
        }
      }
    >
      <Image
        resizeMode="contain"
        source={1}
        style={
          Object {
            "flex": 1,
            "height": undefined,
            "width": undefined,
          }
        }
      />
    </View>
    <FloatingLabelInput
      autoCapitalize="none"
      autoCorrect={false}
      blurOnSubmit={false}
      focusedColor="#55C5CE"
      keyboardType="email-address"
      label=""
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      returnKeyType="next"
      style={
        Object {
          "marginBottom": 15,
        }
      }
      unfocusedColor="#444444"
      value=""
    />
    <View
      style={
        Object {
          "alignSelf": "stretch",
          "marginBottom": 40,
        }
      }
    >
      <FloatingLabelInput
        blurOnSubmit={true}
        focusedColor="#55C5CE"
        label=""
        onChangeText={[Function]}
        onSubmitEditing={[Function]}
        refField={[Function]}
        returnKeyType="done"
        secureTextEntry={true}
        unfocusedColor="#444444"
        value=""
        withRef={true}
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "flex-start",
            "color": "#f54B5E",
            "fontFamily": "Montserrat-Regular",
            "fontSize": 11,
            "marginTop": 2,
          }
        }
      >
         
      </Text>
    </View>
    <Button
      enabled={false}
      onPress={[Function]}
      style={
        Object {
          "marginBottom": 30,
        }
      }
    />
    <TouchableOpacity
      activeOpacity={0.2}
      style={
        Object {
          "marginBottom": 50,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#16ADC1",
            "fontFamily": "Montserrat-Regular",
            "fontSize": 11,
            "textAlign": "center",
            "textDecorationLine": "underline",
          }
        }
      />
    </TouchableOpacity>
  </View>
</TouchableWithoutFeedback>
`;
